packages/smart-queue/src/utils.js

```diff
@@ -0,0 +1,79 @@
+import queueFunctions from './queue-functions';
+
+/**
+ * Returns queue function by method
+ * @param {object} queueMeta
+ * @return {function}
+ */
+export const getQueueFunction = ({ method }) => queueFunctions[method];
+
+/**
+ * Returns an offline action's queue meta
+ * @param {action} action
+ * @return {object}
+ */
+export const getQueueMeta = ({ meta }) => meta.offline.queue;
+
+/**
+ * Asserts whether an offline action's queue metadata is valid
+ * @param {action} queueMeta
+ * @return {boolean}
+ */
+export const valid = queue =>
+  typeof queue === 'object' && !!getQueueFunction(queue) && !!queue.key;
+
+/**
+ * Checks whether an action is safe to be operated on in the queue
+ * if the action is the first in the queue and redux-offline is currently
+ * busy, assume that the action is currently being resolved and is not safe
+ * to operate on
+ * @param {number} index
+ * @param {object} context
+ * @return {boolean}
+ */
+export const safeToProceed = (index, { offline }) =>
+  index !== 0 && !offline.busy;
```

Comment:
@echoes221 ,
I think function `safeToProceed` will return `false` for every case other than `non-zero index` and when `offline is not busy`, but if `offline is not busy` then we can consider `zero` index to be a safe index to merge. Moreover, any `non-zero` index can be considered to be safe irrespective of `offline.busy` status.

@sorodrigo, @echoes221  thoughts?

Reply:
Ah! That should be index===0. Thanks for the catch, I'm still writing tests.